pandas/core/frame.py

```diff
@@ -437,10 +440,12 @@ def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
 
         # For data is list-like, or Iterable (will consume into list)
         elif isinstance(data, abc.Iterable) and not isinstance(data, (str, bytes)):
+            if is_dataclass_instance(data[0]):
+                data = _dataclasses_to_dicts(data)
             if not isinstance(data, abc.Sequence):
                 data = list(data)
             if len(data) > 0:
-                if is_list_like(data[0]) and getattr(data[0], "ndim", 1) == 1:
+                if (is_list_like(data[0]) and getattr(data[0], "ndim", 1) == 1):
```

Comment:
i dont think the linter is going to like this.  run `black pandas` to automatically fix